### ISSUE TITLE
OB/WR mass loss on Post-AGB/WD cooling tracks

### ISSUE THREAD

**Issue by jmerritt1** [TeamCOMPAS/COMPAS#1209]
For stars in the MZAMS range of ~6-9Msol, the `Dominant_Mass_Loss_Type` variable is recording OB(main sequence) and sometimes WR mass loss along white dwarf cooling tracks. The actual mass loss is often small but nonzero after envelope ejection. The specific abnormal mass loss types vary as you change `--mass-change-fraction`, and `--radial-change-fraction`.
<img width="467" alt="Screenshot 2024-08-30 at 11 02 28 AM" src="https://github.com/user-attachments/assets/82abfc63-6efb-4ae5-9622-8299e7fd5f99">
^Color indicates Dominant Mass Loss Rate, not Stellar Type.
<img width="531" alt="Screenshot 2024-08-30 at 11 25 38 AM" src="https://github.com/user-attachments/assets/1f30d2f1-563c-471b-9a43-0539dcb8503f">
DMLR Key:     
    { MASS_LOSS_TYPE::NONE, "NONE" },
    { MASS_LOSS_TYPE::GB,   "GB" },
    { MASS_LOSS_TYPE::LBV,  "LBV" },
    { MASS_LOSS_TYPE::OB,   "OB" },
    { MASS_LOSS_TYPE::RSG,  "RSG" },
    { MASS_LOSS_TYPE::VMS,  "VMS" },
    { MASS_LOSS_TYPE::WR,   "WR" }

<img width="527" alt="Screenshot 2024-08-30 at 11 25 52 AM" src="https://github.com/user-attachments/assets/8fc4bb45-350e-4254-9395-bba68d29241d">
<img width="507" alt="Screenshot 2024-08-30 at 11 26 21 AM" src="https://github.com/user-attachments/assets/422e21ae-cde7-4b67-b068-08e495b9372b">
<img width="616" alt="Screenshot 2024-08-30 at 11 44 00 AM" src="https://github.com/user-attachments/assets/13351257-f7f7-4a15-9be6-051b2abda5d3">

You may not see this behavior with SSE, since evolution is halted at WD formation. With default change fractions the winds go from GB to WR during the post-AGB, and do not dip into MS winds.

**Label the issue**

`urgency_low`      - This issue is not urgent

`severity_minor`    - This is a minor bug with minimal impact

**To Reproduce**
`./COMPAS --mode BSE -n 1 --initial-mass-1 6.5 --initial-mass-2 0.1 --semi-major-axis 1e20 --detailed-output True --mass-change-fraction 0.005 --radial-change-fraction 0.005`

**Expected behavior**
Not sure what to expect, perhaps `NONE` or the new `ZERO` would be most appropriate for WD, and we could set this to be the case for all remnants. For our purposes, a physically realistic way to track mass loss here is probably not important, at least not after the removal of the envelope. It has been difficult to track down why these types (OB, WR) are applied, but the WDs don't seem to have the appropriate Teff, etc. for those winds at those times either.

Also note that we've changed the variable name to `MASS_LOSS_TYPE`, but in the output files it is still recorded as `DOMINANT_MASS_LOSS_RATE`.

**Versioning (please complete the following information):**
 - OS: Ventura 13.6.9
 - COMPAS 3.01.05

**Comment by ilyamandel** [comment]
It also looks like you sometimes have OB winds applied during the blue loop -- is that intentional?

In general, if winds that shouldn't be applied at all are sometimes labeled dominant, it makes me wonder whether they are being applied more often than intended.

According to BaseStar::CalculateMassLossRateFlexible2023(), OB winds are applied all the time, not just to MS stars, as long as the stars are not in the LBV, RSG, cool star, or VMS regimes.  So yes, they will be applied to WDs, and to any other stars not in the regimes above (and in the HURLEY_ADD model, to LBV stars, too).  Is that the intended behaviour?

**Comment by ilyamandel** [comment]
@jmerritt1 , @SimonStevenson : 
Is it the intended behaviour to apply OB winds to all stars that are *not* in the LBV, RSG, cool stars, or VMS regimes (e.g., WDs, but also any other stars that don't satisfy the other criteria, whether or not they would be classified as OB stars)?

**Comment by jmerritt1** [comment]
Thanks, I've opened a PR to set mass loss to 0 for remnants. 

-Regarding blue loops, for the brief time spent above 12.5kK, OB mass loss does seem like the most appropriate treatment we can give, despite their He burning?

-For post AGB mass loss: we don't have a specific prescription, and envelope ejection is forced as per Hurley. It does affect the lifetime and cause a convergence issue depending on timestepping and change fraction settings(mass loss on the asymptotic part is truncated earlier). For example: given 6.5msol at ZAMS, final WD mass is 1.38msol using default settings:

<img width="732" alt="Screenshot 2024-09-05 at 11 18 02 AM" src="https://github.com/user-attachments/assets/a3707f03-1d8d-4216-afae-f88f5abb0173">
<img width="763" alt="Screenshot 2024-09-05 at 11 17 03 AM" src="https://github.com/user-attachments/assets/1245c0cc-e7a4-467e-89b9-ac7d3784adf8">

 but with `--mass-change-fraction 0.005 --radial-change-fraction 0.005`, the final mass is 1.23msol due to the additional time spent losing mass after the initial envelope ejection:
 
<img width="782" alt="Screenshot 2024-09-05 at 11 17 37 AM" src="https://github.com/user-attachments/assets/972d9f09-57ed-43b3-bf37-48a314747a87">
<img width="806" alt="Screenshot 2024-09-05 at 11 17 23 AM" src="https://github.com/user-attachments/assets/7fda55c9-490f-44ec-817f-854798744529">

We may not want OB mass loss stacking with the normal AGB envelope loss, even if the impact is minimal?

**Comment by ilyamandel** [comment]
@jmerritt1 :
I would agree on both points: we probably want OB mass loss during the blue loop, but not as an additive mass loss on the AGB.   
I presume the smaller timesteps on the AGB are bringing us closer to the desired, converged solution, right?

**Comment by ilyamandel** [comment]
@jmerritt1 , @SimonStevenson -- is there a plan to work on this?

**Comment by ilyamandel** [comment]
@jmerritt1 , @SimonStevenson -- is there still an issue here?

**Comment by jmerritt1** [comment]
No plan to work on this currently. 

Checking with COMPAS v03.23.00, there is still OB mass loss on the AGB, though not much. Mass loss is zero for remnants, and there is still OB mass loss on blue loops (which we decided is fine).

**Comment by ilyamandel** [comment]
OK, it seems that things are mostly behaving as expected now that remnants no longer experience winds.  Note also the fix in https://github.com/TeamCOMPAS/COMPAS/pull/1433 for non-naked-He stars that experience WR winds.  I'm therefore closing this issue.
